gravitee-apim-rest-api/gravitee-apim-rest-api-automation/gravitee-apim-rest-api-automation-rest/src/test/java/io/gravitee/apim/rest/api/automation/spring/PermissionsFilterTest.java

```diff
@@ -0,0 +1,268 @@
+import io.gravitee.rest.api.service.PermissionService;
+import io.gravitee.rest.api.service.common.GraviteeContext;
+import io.gravitee.rest.api.service.common.IdBuilder;
+import io.gravitee.rest.api.service.exceptions.ForbiddenAccessException;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.core.MultivaluedHashMap;
+import jakarta.ws.rs.core.UriInfo;
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/**
+ * @author Kamiel Ahmadpour (kamiel.ahmadpour at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class PermissionsFilterTest {
+
+    @InjectMocks
+    protected PermissionsFilter permissionFilter;
+
+    @Mock
+    protected PermissionService permissionService;
+
+    @Mock
+    protected Permissions permissions;
+
+    @Mock
+    protected ContainerRequestContext containerRequestContext;
+
+    private static final String HRID = "test-hrid";
+    private static final String HRID_PARAM = "hrid";
+    private static final String LEGACY_PARAM = "legacy";
+    private static final String ORGANIZATION_ID = "ORG_ID";
+    private static final String ENVIRONMENT_ID = "DEFAULT";
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        GraviteeContext.setCurrentOrganization(ORGANIZATION_ID);
+        GraviteeContext.setCurrentEnvironment(ENVIRONMENT_ID);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        GraviteeContext.cleanContext();
+    }
+
+    @Nested
+    class Api {
+
+        @BeforeEach
+        void initApiMock() {
+            Permission perm = mock(Permission.class);
+            when(perm.value()).thenReturn(API_DEFINITION);
+            when(perm.acls()).thenReturn(new RolePermissionAction[] { READ });
+            when(permissions.value()).thenReturn(new Permission[] { perm });
+            UriInfo uriInfo = mock(UriInfo.class);
+            MultivaluedHashMap<String, String> map = new MultivaluedHashMap<>();
+            map.put(HRID_PARAM, Collections.singletonList(HRID));
+            when(uriInfo.getPathParameters()).thenReturn(map);
+            MultivaluedHashMap<String, String> queryParams = new MultivaluedHashMap<>();
+            queryParams.put(LEGACY_PARAM, Collections.singletonList("true"));
+            when(uriInfo.getQueryParameters()).thenReturn(queryParams);
+            when(containerRequestContext.getUriInfo()).thenReturn(uriInfo);
+            when(containerRequestContext.getMethod()).thenReturn("GET");
+        }
+
+        @Test
+        void shouldThrowForbiddenExceptionWhenNoApiPermissions() {
+            when(permissionService.hasPermission(any(), any(), any())).thenReturn(false);
```

Comment:
maybe we could use the actual permission in a eq() here at least to make the test more explanatory ?